### PR TITLE
Use a simple diagnostic identifier to dedupe diagnostics

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs [hunk 1]
```diff
@@ -419,12 +419,20 @@ private ImmutableArray<LSP.Diagnostic> ConvertDiagnostic(IDiagnosticSource diagn
...
-                        .GetHashCode().ToString();
+                    // Note: LSP client buckets diagnostics against each file uri.  So we do not need to include any
+                    // file information in this identifier.
+                    //
+                    // Note: It's possible that different systems might report a slightly different diagnostic.  For
+                    // example potentially a slightly different error message from the compiler.  Generally, we think
+                    // this should be very unlikely, and that by just having the right code and location for the
+                    // diagnostic, resolving the issue should always be manageable regardless of which diagnostic
+                    // message is being shown.
+                    diagnostic.Identifier = FormattableString.Invariant($"{diagnostic.Code}-{diagnostic.Range.Start.Line}-{diagnostic.Range.Start.Character}");
```
Comment (CyrusNajmabadi) [inline src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs:435]: Note the use of 'Invariant' so we're insulated from any sort of culture concerns.

**Review by adrianvmsft — APPROVED** [review]

**Review by kayle — APPROVED** [review]

File: src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs [hunk 2]
```diff
@@ -419,12 +419,20 @@ private ImmutableArray<LSP.Diagnostic> ConvertDiagnostic(IDiagnosticSource diagn
...
```
Comment (sharwell) [inline src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs:435]: The compiler reports multiple instances of CS8618 at the same location (the constructor identifier) for failing to initialize a non-nullable field. For example, the following contains two diagnostics (one for `value` and one for `value2`):

```cs
class Example
{
    string value;
    string value2;

    Example() { }
}
```

Comment (CyrusNajmabadi) [reply src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs:435]: hrmm.. that def throws a wrinkle into things :(

**Comment by CyrusNajmabadi** [comment]
Closing out as this won't work.

**Pull request closed without merging**